client/my-sites/promote-post/test/credit-balance.test.tsx

```diff
@@ -0,0 +1,49 @@
+/**
+ * @jest-environment jsdom
+ */
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+import { BlazeCreditStatus, useBlazeCredits } from 'calypso/lib/promote-post';
+import CreditBalance from '../components/credit-balance';
+
+type UseBlazeCreditsMock = jest.Mock< BlazeCreditStatus >;
+
+jest.mock( 'calypso/lib/promote-post', () => {
+	const module = jest.requireActual( 'calypso/lib/promote-post' );
+	return {
+		...module,
+		useBlazeCredits: jest.fn< UseBlazeCreditsMock, [] >(),
+	};
+} );
+
+describe( 'CreditBalance component', () => {
+	test( 'displays null when balance is 0', () => {
+		const { container } = render( <CreditBalance /> );
+		expect( container.firstChild ).toBeNull();
+	} );
+
+	describe( 'CreditBalance component when balance is set', () => {
+		const mockBalance = 10;
+
+		beforeEach( () => {
+			jest.clearAllMocks();
+
+			const useStateSpy = jest.spyOn( React, 'useState' );
+			useStateSpy.mockReturnValueOnce( [ mockBalance, jest.fn() ] );
+		} );
+
+		test( 'displays null when credits are not enabled for the user', () => {
+			( useBlazeCredits as UseBlazeCreditsMock ).mockReturnValue( BlazeCreditStatus.DISABLED );
```

Comment:
Had to do this to avoid TS errors `TS2339: Property 'mockReturnValue' does not exist on type '() => BlazeCreditStatus'.`